library/std/Cargo.toml

```diff
@@ -17,7 +17,7 @@ cfg-if = { version = "1.0", features = ['rustc-dep-of-std'] }
 panic_unwind = { path = "../panic_unwind", optional = true }
 panic_abort = { path = "../panic_abort" }
 core = { path = "../core", public = true }
-compiler_builtins = { version = "0.1.105" }
+compiler_builtins = { version = "0.1.112", features = ["no-f16-f128"] }
```

Comment:
So if someone only builds `core`, they still get the feature? That seems problematic?

Reply:
Please instead add a `compiler-builtins-no-f16-f128` feature to liballoc, libstd and sysroot which enables `no-f16-f128` and then enable this feature at https://github.com/rust-lang/rust/blob/b71fa82d786ae1b5866510f1b3a7e5b7e1890e4c/compiler/rustc_codegen_cranelift/build_system/build_sysroot.rs#L275 and https://github.com/rust-lang/rust/blob/b71fa82d786ae1b5866510f1b3a7e5b7e1890e4c/compiler/rustc_codegen_gcc/build_system/src/build.rs#L115 for cg_clif and cg_gcc respectively.

Reply:
I've made this change and pushed. However, when I tested with `ENABLE_GCC_CODEGEN=1 src/ci/docker/run.sh x86_64-gnu-llvm-17` it still failed with a panic related to f16/128. I don't think it ran the code in `rust/compiler/rustc_codegen_gcc/build_system/src/build.rs`. Is there another step I'm missing?

Reply:
Unfortunately the current situation is pretty bad. [A PR](https://github.com/rust-lang/rust/pull/124353) to improve things is open and currently waiting from infra team information. Maybe give it a try?